internal/compiler/passes/resolving.rs

```diff
@@ -1139,6 +1127,123 @@ fn maybe_lookup_object(
     base
 }
 
+/// Go through all the two way binding and resolve them first
+fn resolve_two_way_bindings(
+    doc: &Document,
+    type_register: &TypeRegister,
+    diag: &mut BuildDiagnostics,
+) {
+    for component in doc.inner_components.iter() {
+        let scope = ComponentScope(vec![]);
+
+        recurse_elem(&component.root_element, &scope, &mut |elem, scope| {
+            let mut new_scope = scope.clone();
+            new_scope.0.push(elem.clone());
+            for (prop_name, binding) in &elem.borrow().bindings {
+                let mut binding = binding.borrow_mut();
+                if let Expression::Uncompiled(node) = binding.expression.clone() {
+                    if let Some(n) = syntax_nodes::TwoWayBinding::new(node.clone()) {
+                        let lhs_lookup = elem.borrow().lookup_property(prop_name);
+                        if lhs_lookup.property_type == Type::Invalid {
+                            // An attempt to resolve this already failed when trying to resolve the property type
+                            assert!(diag.has_error());
```

Comment:
Maybe debug_assert?

Reply:
I'm going for an hard assert here because it is a cheap operation, and we really should have an error, so even in release mode it is better to assert to make it an error rather than allowing invalid code and continue.